### PR TITLE
fix *crud fro cross_reference, editor, mesh_detail, reference

### PR DESCRIPTION
**Pull request by FB-ping**
I fixed *crud patch bug for cross_reference, reference, editor, mesh_detail, still need to fix for mod_reference_type. There is no data for file, note, people (same as person ?), reference_automated_term_tags, reference_manul_term_tags, so I have not tested this yet.

### CODE REVIEW

**Comment by azurebrd** [comment]
@FB-ping I don't know much about this, so hopefully Adam can review it.

File: backend/app/literature/crud/editor_crud.py [hunk 1]
```diff
@@ -50,15 +50,15 @@ def destroy(db: Session, editor_id: int) -> None:
...
+    editor_data = jsonable_encoder(editor_update)
     editor_db_obj = db.query(EditorModel).filter(EditorModel.editor_id == editor_id).first()
     if not editor_db_obj:
         raise HTTPException(status_code=status.HTTP_404_NOT_FOUND,
                             detail=f"Editor with editor_id {editor_id} not found")
     res_ref = stripout(db, editor_update, non_fatal=True)
     add(res_ref, editor_db_obj)
 
-    for field, value in editor_update.dict().items():
+    for field, value in editor_data.items():
```
Comment (adamjohnwright) [inline backend/app/literature/crud/editor_crud.py:61]: I believe @dougli1sqrd added in the .dict() when he was doing the large MyPy pull request. I thought it was not necessary. You should consult with him to make sure he agrees.

File: backend/app/literature/crud/mesh_detail_crud.py [hunk 2]
```diff
@@ -43,12 +43,13 @@ def destroy(db: Session, mesh_detail_id: int) -> None:
 
 
 def patch(db: Session, mesh_detail_id: int, mesh_detail_update: MeshDetailSchemaUpdate) -> dict:
+    mesh_detail_data = jsonable_encoder(mesh_detail_update)
```
Comment (adamjohnwright) [inline backend/app/literature/crud/mesh_detail_crud.py:46]: I would also check with @ianlongden  to make sure that the patch functions work with the generic update functions that he made. I am not sure if we can use his functions in this context. Or maybe we should be extending what he has already done. I think his functions are only for adding things related to reference/resource.

**Review by adamjohnwright — APPROVED** [review]
Everything looks good as long as @ianlongden and @dougli1sqrd agree with the changes

File: backend/app/literature/crud/editor_crud.py (hunk 1 above)
Comment (dougli1sqrd) [reply backend/app/literature/crud/editor_crud.py:61]: If mypy checks out (which it seems like it does) and this runs just fine then this is fine by me. I think I was previously getting errors that items() is not a function on that type. But this seems fine by me!

**Review by dougli1sqrd — APPROVED** [review]

**Review by ianlongden — APPROVED** [review]

**Pull request merged**